src/nools/lib.js

```diff
@@ -123,13 +123,14 @@ function emitTasksForSchedule(c, schedule, r) {
   }
 
   function initActions(def) {
+    var appliesToReport = !!r;
     var content = {
       source: 'task',
-      source_id: r && r._id,
+      source_id: appliesToReport ? r._id : c.contact && c.contact._id,
       contact: c.contact,
     };
 
-    if(def.modifyContent) def.modifyContent(r, content);
+    if(def.modifyContent) def.modifyContent(content, c, r);
```

Comment:
It appears that you're changing the parameter order for this function, which may break existent configurations?
It also differs from the the documented version: https://github.com/medic/medic-docs/blob/master/configuration/developing-community-health-applications.md#tasksjs

Reply:
Yep